src/corecel/math/ArrayUtils.hh

```diff
@@ -198,6 +204,33 @@ is_soft_orthogonal(Array<T, N> const& x, Array<T, N> const& y)
     return soft_zero(dot_product(x, y));
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Check whether two vectors are approximately collinear.
+ *
+ * This is done by checking if the Gram determinant is soft zero:
+ * \f[
+ * (x \cdot x) (y \cdot y) - (x \cdot y)^2 \approx 0
+ * \f]
+ * This method is advantageous because it doesn't require normalizing inputs.
+ */
+template<class T, size_type N>
+inline CELER_FUNCTION bool
+is_soft_collinear(Array<T, N> const& x, Array<T, N> const& y)
+{
+    SoftZero const soft_zero{SoftEqual{}.rel()};
```

Comment:
`SoftEqual` needs to explicitly use type `<T>`, not the default `real_type`, for the single-precision build to pass. The construction generator I think should work automatically:
```suggestion
    SoftZero const soft_zero{SoftEqual<T>{}.rel()};
```